packages/client/src/pages/Bots/BotForm.re

```diff
@@ -0,0 +1,93 @@
+module FormConfig = {
+  type field(_) =
+    | Name: field(string)
+    | Description: field(string)
+    | CompetitorType: field(Context.Primary.CompetitorType.t);
+
+  type state = {
+    name: string,
+    description: string,
+    competitorType: Context.Primary.CompetitorType.t,
+  };
+
+  let get: type value. (state, field(value)) => value =
+    (state, field) =>
+      switch (field) {
+      | Name => state.name
+      | Description => state.description
+      | CompetitorType => state.competitorType
+      };
+
+  let set: type value. (state, field(value), value) => state =
+    (state, field, value) =>
+      switch (field) {
+      | Name => {...state, name: value}
+      | Description => {...state, description: value}
+      | CompetitorType => {...state, competitorType: value}
+      };
+};
+
+module Form = ReFormNext.Make(FormConfig);
+
+let withForm =
+    (onSubmit, bot: option(Context.Primary.Bot.t), innerComponentFn) => {
+  let initialState: FormConfig.state =
+    switch (bot) {
+    | Some(bot) => {
+        name: bot.name |> Rationale.Option.default(""),
```

Comment:
Can you use ``E.O.default`` or similar instead? I'd prefer not to reference Rationale directly, because it's confusing with three different libraries (Rationale, Belt, StdLib)